### PR TITLE
[#5118] feat(auth):  Lakehouse Iceberg catalog supports Ranger authorization plugin

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?

Lakehouse Iceberg catalog supports Ranger authorization plugin

### Why are the changes needed?

Fix: #5118 

### Does this PR introduce _any_ user-facing change?

Add the document.

### How was this patch tested?

Add E2E tests.

### CODE REVIEW

File: authorizations/authorization-ranger/build.gradle.kts [hunk 1]
```diff
@@ -126,7 +129,7 @@ tasks {
 
 tasks.test {
   doFirst {
-    environment("HADOOP_USER_NAME", "test")
+    environment("HADOOP_USER_NAME", "gravitino")
   }
```
Comment (xunliu) [inline authorizations/authorization-ranger/build.gradle.kts:133]: The `HADOOP_USER_NAME` default is `gravitino`, If we do not need to set a different value, we can remove this paragraph code.

File: authorizations/authorization-ranger/src/test/java/org/apache/gravitino/authorization/ranger/integration/test/RangerIcebergE2EIT.java
Comment (xunliu) [inline authorizations/authorization-ranger/src/test/java/org/apache/gravitino/authorization/ranger/integration/test/RangerIcebergE2EIT.java:1]: The `RangerIcebergE2EIT` class is the same as `RangerHiveE2EIT`, I think we can abstract a common IT class.

Comment (jerqi) [reply authorizations/authorization-ranger/src/test/java/org/apache/gravitino/authorization/ranger/integration/test/RangerIcebergE2EIT.java:1]: They are similar but different.  For same update SQL, Iceberg will succeed but Hive will throw exception.

File: authorizations/authorization-ranger/build.gradle.kts (hunk 1 above)
Comment (jerqi) [reply authorizations/authorization-ranger/build.gradle.kts]: OK, reverted.

Comment (jerqi) [reply authorizations/authorization-ranger/build.gradle.kts]: Default HADOOP_HOME is anoymous. So I added it back.

File: docs/security/authorization-pushdown.md [hunk 2]
```diff
@@ -43,6 +43,36 @@ authorization.ranger.password=PWD123
...
+| `authorization.ranger.admin.url`    | The Apache Ranger web URIs.                                                                                                                          | (none)        | No       | 0.6.0-incubating |
+| `authorization.ranger.auth.type`    | The Apache Ranger authentication type `simple` or `kerberos`.                                                                                        | `simple`      | No       | 0.6.0-incubating |
+| `authorization.ranger.username`     | The Apache Ranger admin web login username (auth type=simple), or kerberos principal(auth type=kerberos), Need have Ranger administrator permission. | (none)        | No       | 0.6.0-incubating |
+| `authorization.ranger.password`     | The Apache Ranger admin web login user password (auth type=simple), or path of the keytab file(auth type=kerberos)                                   | (none)        | No       | 0.6.0-incubating |
+| `authorization.ranger.service.name` | The Apache Ranger service name.                                                                                                                      | (none)        | No       | 0.6.0-incubating |
```
Comment (xunliu) [inline docs/security/authorization-pushdown.md:57]: Maybe need to change `0.6.0-incubating` to `0.8.0-xxx`?

Comment (jerqi) [reply docs/security/authorization-pushdown.md:57]: This option reuse Hive properties. I don't a a new properity. This propeties are introduced in 0.6.0-incubating.

Comment (jerqi) [reply docs/security/authorization-pushdown.md]: Change to 0.8.0-incubating.

**Review by xunliu — APPROVED** [review]
LGTM

**Pull request merged**